crates/biome_service/src/workspace.rs

```diff
@@ -977,13 +996,31 @@ pub trait Workspace: Send + Sync + RefUnwindSafe {
         params: SetManifestForProjectParams,
     ) -> Result<(), WorkspaceError>;
 
-    /// Register a possible workspace project folder. Returns the key of said project. Use this key when you want to switch to different projects.
+    /// Registers a possible workspace project folder.
+    ///
+    /// Returns the key of said project. Use this key when you want to switch to different projects.
     fn register_project_folder(
         &self,
         params: RegisterProjectFolderParams,
     ) -> Result<ProjectKey, WorkspaceError>;
 
-    /// Unregister a workspace project folder. The settings that belong to that project are deleted.
+    /// Scans the project folder and populates service data.
+    ///
+    /// The first time you call this method, it may take a long data since it will traverse the
+    /// entire project folder recursively, parse all included files (and possibly their
+    /// dependencies), and perform processing for extracting the service data.
+    ///
+    /// Follow-up calls may be much faster as they can reuse cached data.
+    ///
+    /// TODO: This method also registers file watchers to make sure the cache remains up-to-date.
+    fn scan_project_folder(&self) -> Result<ScanProjectFolderResult, WorkspaceError>;
```

Comment:
This method needs to accept a path to tell the workspace **which** project to scan. 

The LSP can register multiple workspace folders. 

EDIT: This method scans the current project, but the comments didn't make that clear. We should at least say that.